File: Source/Jekov.Nevix/Common/Jekov.Nevix.Common.ViewModels/MediaFolderViewModel.cs
namespace Jekov.Nevix.Common.ViewModels
{
    using Jekov.Nevix.Common.Models;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.Runtime.Serialization;
    using System.Security.Cryptography;
    using System.Text;

    [DataContract]
    public class MediaFolderViewModel
    {
        public MediaFolderViewModel()
        {
            this.Folders = new List<MediaFolderViewModel>();
            this.Files = new List<MediaFileViewModel>();
        }

        [Required]
        [DataMember(Name = "name")]
        [StringLength(ModelConstants.NameLength)]
        public string Name { get; set; }

        [StringLength(ModelConstants.LocationLength)]
        [DataMember(Name = "location")]
        public string Location { get; set; }

        [DataMember(Name = "folders")]
        public ICollection<MediaFolderViewModel> Folders { get; set; }

        [DataMember(Name = "files")]
        public ICollection<MediaFileViewModel> Files { get; set; }
        
        public string GetAllLocations()
        {
            StringBuilder sb = new StringBuilder();
            sb.Append(this.Location);
            foreach (var file in this.Files)
            {
                sb.Append(file.Location);
            }
            foreach (var subFolder in this.Folders)
            {
                sb.Append(subFolder.GetAllLocations());
            }

            return sb.ToString();
        }
    }
}

File: Source/Android/Nevix/nevix/src/main/java/com/antonyjekov/nevix/viewmodels/MediaFolderViewModel.java
package com.antonyjekov.nevix.viewmodels;

import java.util.ArrayList;

public class MediaFolderViewModel {
    private String name;
    private ArrayList<MediaFolderViewModel> folders;
    private ArrayList<MediaFileViewModel> files;

    public MediaFolderViewModel() {
        this.folders = new ArrayList<MediaFolderViewModel>();
        this.files = new ArrayList<MediaFileViewModel>();
    }

    public ArrayList<MediaFolderViewModel> getFolders() {
        return folders;
    }

    public void setFolders(ArrayList<MediaFolderViewModel> folders) {
        this.folders = folders;
    }

    public ArrayList<MediaFileViewModel> getFiles() {
        return files;
    }

    public void setFiles(ArrayList<MediaFileViewModel> files) {
        this.files = files;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}
